var/spack/repos/builtin/packages/r/package.py

```diff
@@ -86,29 +87,40 @@ class R(Package):
     def etcdir(self):
         return join_path(prefix, 'rlib', 'R', 'etc')
 
-    def install(self, spec, prefix):
-        rlibdir = join_path(prefix, 'rlib')
-        configure_args = ['--prefix=%s' % prefix,
-                          '--libdir=%s' % rlibdir,
-                          '--enable-R-shlib',
-                          '--enable-BLAS-shlib',
-                          '--enable-R-framework=no']
+    def configure_args(self):
+        spec   = self.spec
+        prefix = self.prefix
+
+        config_args = [
+            '--libdir={0}'.format(join_path(prefix, 'rlib')),
+            '--enable-R-shlib',
+            '--enable-BLAS-shlib',
+            '--enable-R-framework=no'
+        ]
+
         if '+external-lapack' in spec:
-            configure_args.extend(['--with-blas', '--with-lapack'])
+            config_args.extend([
+                '--with-blas',
+                '--with-lapack'
+            ])
 
-        configure(*configure_args)
-        make()
-        make('install')
+        if '+X' in spec:
+            config_args.append('--with-x')
+        else:
+            config_args.append('--without-x')
```

Comment:
See the main conversation thread about building R without X on linux systems and R's use of Tk.